### PR TITLE
First iteration move

### PR DESCRIPTION
**Pull request by Tjofil**
First iteration move:
- Add dependency-leaf classes (No modifications to original ones from PA-RCA repo).
- Setup initial gradle dependencies.
- Introduce version.properties versioning from core.
- Add spotbugs gradle tasks and settings.
- Adjust some gradle tasks.
- Update gradle to jdk 19 compatible version (7.6.1).
- Add spotlessApply for auto-formatting.

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/performance-analyzer-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by khushbr** [comment]
@Tjofil Thank you for adding these changes, few points to add:
1. We don't need to move code exclusively consumed by Reader (/rca and /reader sub-directory) to the commons repo. 
2. Any reason we are using JDK19 ? I see in PA and RCA, we build for JDK11 and JDK17.

**Comment by khushbr** [comment]
Hey @Tjofil , Can we flatten the directory structure here and organize the classes in more modular manner. Here is one suggestion:

```
...
--\src
----\src \main
------\src \main \config \ConfigStatus.java
------\src \main \config \ConfigFatalException.java
------\src \main \config \ConfigFileException.java
------\src \main \config \PluginSettings.java
------\src \main \config \SettingsHelper.java
------\src \main \util \Util.java
-------\src \main \metrics \AllMetrics.java
-------\src \main \metrics \MetricDimension.java
-------\src \main \metrics \MetricValue.java
-------\src \main \metrics \WriterMetrics.java
-------\src \main \framework \JooqFieldValue.java
---------\src \main \framework \stats \Statistics.java
---------\src \main \framework \stats \Formatter.java
------\src \main \framework \MeasurementSet.java
------\src \main \event_process \EventDispatcher.java
-------\src \main \event_process \EventProcessor.java
-------\src \main \event_process \Event.java
...
```

**Review by khushbr — APPROVED** [review]
LGTM! Thank you for making the changes, @Tjofil !

**Review by ansjcy — APPROVED** [review]

File: src/main/java/org/opensearch/performanceanalyzer/commons/event_process/EventDispatcher.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+package org.opensearch.performanceanalyzer.commons.event_process;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class EventDispatcher {
```
Comment (khushbr) [inline src/main/java/org/opensearch/performanceanalyzer/commons/event_process/EventDispatcher.java:14]: The `EventDispatcher` class should be part of the RCA codebase, any reason we have included it here?

File: src/main/java/org/opensearch/performanceanalyzer/commons/event_process/EventProcessor.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.performanceanalyzer.commons.event_process;
+
+public interface EventProcessor {
```
Comment (khushbr) [inline src/main/java/org/opensearch/performanceanalyzer/commons/event_process/EventProcessor.java:8]: Same as above for the `EventProcessor` class.

File: src/main/java/org/opensearch/performanceanalyzer/commons/event_process/EventDispatcher.java (hunk 1 above)
Comment (Tjofil) [reply src/main/java/org/opensearch/performanceanalyzer/commons/event_process/EventDispatcher.java:14]: 'EventDispatcher' and 'EventProcessor' are direct dependencies of 'EventLogFileHandler' that is to come in next PR. They're here now as they don't have tests on their own and not to overwhelm future PR's which may be hardver to review.

Comment (khushbr) [reply src/main/java/org/opensearch/performanceanalyzer/commons/event_process/EventDispatcher.java:14]: Thanks for clarifying this @Tjofil , I looked deeper in the code, and it makes sense to move `EventLogFileHandler` and the common Event classes to this repo.

**Review by khushbr — APPROVED** [review]

**Review by ansjcy — APPROVED** [review]

**Pull request merged**